fedimint-server/src/lib.rs

```diff
@@ -90,11 +92,28 @@ impl FedimintServer {
     ) -> anyhow::Result<()> {
         let server = FedimintServer::new(cfg.clone(), consensus, decoders, task_group).await;
         let server_consensus = server.consensus.clone();
+        let consensus_hash = server
+            .cfg
+            .consensus
+            .to_config_response(&server_consensus.module_inits)
+            .consensus_hash;
+
         task_group
             .spawn("api-server", |handle| {
                 net::api::run_server(cfg, server_consensus, handle)
             })
             .await;
+
+        loop {
+            info!("Waiting for peers to agree on a consensus config hash");
+            match server.api.download_client_config().await {
+                Ok(response) if response.consensus_hash == consensus_hash => break,
+                Ok(_) => bail!("Our consensus config doesn't match peers!"),
+                Err(_) => {}
+            }
+            sleep(Duration::from_millis(1000)).await;
+        }
```

Comment:
Isn't there a query strategy for this?

Reply:
The problem is when the federation first starts they all "agree" on the endpoint being down.

I could try to modify the Retry404 to be more generic in ignoring errors.

Reply:
"We are all down, so we stay all down!" :laughing: